packages/commerce-sdk-react/src/hooks/ShopperCustomers/mutation.test.tsx

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React from 'react'
+import path from 'path'
+import {mockHttpResponses, renderWithProviders, queryClient} from '../../test-utils'
+import {fireEvent, screen, waitFor} from '@testing-library/react'
+import {ShopperLoginHelpers, useShopperLoginHelper} from '../ShopperLogin'
+import {useShopperCustomersMutation, ShopperCustomersMutations} from './mutation'
+import nock from 'nock'
+
+const {withMocks} = mockHttpResponses({directory: path.join(__dirname, '../../../mock-responses')})
+
+const CUSTOMER_EMAIL = 'kobe@test.com'
+const CUSTOMER_ID = 'abkehFwKoXkbcRmrFIlaYYwKtJ'
+const RANDOM_STR = Math.random()
+    .toString(36)
+    .slice(2, 7)
+
+const CustomerMutationComponent = () => {
+    const loginRegisteredUser = useShopperLoginHelper(ShopperLoginHelpers.LoginRegisteredUserB2C)
+    const updateCustomer = useShopperCustomersMutation(ShopperCustomersMutations.UpdateCustomer)
+
+    React.useEffect(() => {
+        loginRegisteredUser.mutate({
+            username: 'kobe@test.com',
+            password: 'Test1234!'
+        })
+    }, [])
+
+    return (
+        <>
+            {loginRegisteredUser.isLoading ? (
+                <span>Logging in...</span>
+            ) : (
+                <div>Logged in as {loginRegisteredUser?.variables?.username}</div>
+            )}
+            <div>
+                <button
+                    onClick={() =>
+                        updateCustomer.mutate({
+                            parameters: {customerId: CUSTOMER_ID},
+                            body: {firstName: `Kobe${RANDOM_STR}`}
+                        })
+                    }
+                >
+                    Update Customer
+                </button>
+
+                {updateCustomer.error?.message && (
+                    <p style={{color: 'red'}}>Error: {updateCustomer.error?.message}</p>
+                )}
+                <hr />
+                <div>{JSON.stringify(updateCustomer)}</div>
+            </div>
+        </>
+    )
+}
+
+const tests = [
+    {
+        hook: 'updateCustomer',
+        cases: [
+            {
+                name: 'success',
+                assertions: withMocks(async () => {
```

Comment:
We can remove this `withMocks` if we don't plan on recording the http responses via `nockBack`. But then we'll need to call `mockAuthCalls` from test-utils.tsx file.

So the end result is something like:
```
mockAuthCalls()

nock('http://localhost:3000')
                         .patch((uri) => {
                             return uri.includes('/customer/shopper-customers/')
                         })
```

Reply:
Thanks @vmarta for paring on implementing this change.

The mutation tests are now passing without the need of recordings!